static/app/components/globalDrawer/index.spec.tsx

```diff
@@ -0,0 +1,198 @@
+    const closeSpy = jest.fn();
+
+    render(
+      <GlobalDrawerTestComponent
+        config={{
+          renderer: ({Body}) => (
+            <Body data-test-id="drawer-test-content">onClose outside click</Body>
+          ),
+          options: {onClose: closeSpy},
+        }}
+      />
+    );
+
+    await userEvent.click(screen.getByTestId('drawer-test-open'));
+
+    expect(await screen.findByTestId('drawer-test-content')).toBeInTheDocument();
+
+    await userEvent.click(screen.getByTestId('drawer-test-outside'));
+
+    expect(closeSpy).toHaveBeenCalled();
+    expect(screen.queryByTestId('drawer-test-content')).not.toBeInTheDocument();
+  });
+
+  // TODO(Leander): I'll have to look into this further in a separate PR.
+  // The useHotKeys hook doesn't seem to register the userEvents for this test
+  it.skip('calls onClose handler when escape key is pressed', async function () {
+    const closeSpy = jest.fn();
+
+    render(
+      <GlobalDrawerTestComponent
+        config={{
+          renderer: ({Body}) => (
+            <Body data-test-id="drawer-test-content">onClose escape</Body>
+          ),
+          options: {onClose: closeSpy},
+        }}
+      />
+    );
+
+    await userEvent.click(screen.getByTestId('drawer-test-open'));
+
+    const content = screen.getByTestId('drawer-test-content');
+    expect(content).toBeInTheDocument();
+
+    await userEvent.keyboard('{Escape}');
+
+    expect(closeSpy).toHaveBeenCalled();
+    expect(content).not.toBeInTheDocument();
+  });
+
+  it('calls onClose handler when closeDrawer prop is called', async function () {
+    const closeSpy = jest.fn();
+
+    render(
+      <GlobalDrawerTestComponent
+        config={{
+          renderer: ({closeDrawer: cd}) => (
+            <button data-test-id="drawer-test-content" onClick={cd}>
+              onClose prop
+            </button>
+          ),
+          options: {onClose: closeSpy},
+        }}
+      />
+    );
+
+    await userEvent.click(screen.getByTestId('drawer-test-open'));
+
+    const button = screen.getByTestId('drawer-test-content');
+    expect(button).toBeInTheDocument();
+
+    await userEvent.click(button);
+
+    expect(closeSpy).toHaveBeenCalled();
+    expect(button).not.toBeInTheDocument();
+  });
+
+  it('ignores some close events press when option is set', async function () {
+    const closeSpy = jest.fn();
+
+    render(
+      <GlobalDrawerTestComponent
+        config={{
+          renderer: ({Body}) => (
+            <Body data-test-id="drawer-test-content">ignore close events</Body>
+          ),
+          options: {
+            onClose: closeSpy,
+            closeOnEscapeKeypress: false,
+            closeOnOutsideClick: false,
+          },
+        }}
+      />
+    );
+
+    await userEvent.click(screen.getByTestId('drawer-test-open'));
+
+    const content = screen.getByTestId('drawer-test-content');
+
+    await userEvent.keyboard('{Escape}');
```

Comment:
Maybe drop a todo here to fix whatever needs fixed when the useHotKey stuff is fixed